packages/routing-utils/src/types.ts

```diff
@@ -25,6 +25,37 @@ export type ConditionValue =
       lte?: number;
     };
 
+export interface MitigateRateLimit {
+  algo: 'fixed_window' | 'token_bucket';
+  window: number;
+  limit: number;
+  keys: string[];
+  action?: 'log' | 'challenge' | 'deny' | 'rate_limit' | null;
+}
+
+export interface MitigateRedirect {
+  location: string;
+  permanent: boolean;
+}
+
+export type MitigateActionType =
+  | 'log'
+  | 'challenge'
+  | 'deny'
+  | 'bypass'
+  | 'rate_limit'
+  | 'redirect';
```

Comment:
I think we can remove `redirect` here since we already have `redirects` top level in vercel.json